### DISCUSSION TITLE
Python3.7 Lib error after upgrading

### DISCUSSION THREAD

**Discussion by grootum** [apache/superset#17802] [Q&A / Help]
Hi,

everything works perfectly fine, but recently I pulled changes in repo and then I ran into a problem with initialization Superset frontend via docker-compose

The superset_app container is stuck on this Python error
`ImportError: cannot import name 'TypedDict' from 'typing' (/usr/local/lib/python3.7/typing.py)`

Can you help me, what could be wrong?

Thanks

**Comment by villebro** [comment]
@grootum A few months ago we added support for Python 3.9, and sent a notice to the mailing list that support for Python 3.7 will soon be dropped: https://lists.apache.org/thread/v8490vf1horzcy7h5f8oh6xdxh0z2rb2 . I would recommend upgrading to Python 3.8, as that's what we're currently running on CI.

**Reply by grootum** [reply]
Sorry, but I dont get it. 
I am using installation via Docker Compose, should I exec to container then and upgrade Python there or what?

**Reply by villebro** [reply]
I opened a PR to fix this issue: https://github.com/apache/superset/pull/17823 . Once that gets merged I believe this error should go away. If you reinitialize your docker-compose env you should get an environment with Python 3.8 (it appears your docker-compose has been initialized when Python 3.7 was still being used).

**Reply by grootum** [reply]
Thanks, mate, everything works perfectly now.

**Reply by klazaj** [reply]
@villebro what do you mean by reinitialize the docker-compose? I just upgraded from Superset 1.1.0 to Superset 1.5.0, and am getting the same error. It seems the Python version is still 3.7, and is not upgraded to 3.9.
